aspnet/mvc/models/validation.rst

```diff
@@ -1,8 +1,196 @@
+
+MVC uses data annotations in addition to type metadata from model properties to validate data and display any error messages using JavaScript. When you use MVC to render form elements from a model using `Tag Helpers <https://docs.asp.net/en/latest/mvc/views/tag-helpers/index.html>`_ or `HTML helpers <https://docs.asp.net/en/latest/mvc/views/html-helpers.html>`_ it will add HTML 5 `data- attributes. <http://w3c.github.io/html/dom.html#embedding-custom-non-visible-data-with-the-data-attributes>`_ in the form elements that need validation, as shown below. MVC generates the ``data-`` attributes for both built-in and custom attributes. The ``data-val-required`` attribute below contains an error message to display if the user doesn't fill in the release date field, and that message displays in the accompanying ``<span>`` element. You can display validation errors on the client using the relevant tag helpers as shown here:
+
+.. code-block:: html
+ :emphasize-lines: 5, 10
+
+  <form asp-action="Create">
+    <div class="form-horizontal">
+      <h4>Movie</h4>
+      <hr />
+      <div asp-validation-summary="ValidationSummary.ModelOnly" class="text-danger"></div>
+        <div class="form-group">
+          <label asp-for="ReleaseDate" class="col-md-2 control-label"></label>
+          <div class="col-md-10">
+            <input asp-for="ReleaseDate" class="form-control" />
+            <span asp-validation-for="ReleaseDate" class="text-danger" />
+          </div>		
+      </div>	
+    </div>  
+  </form>
+
+The tag helper above renders the HTML below. Notice that the ``data-`` attributes in the HTML output correspond to the data annotations for the ``ReleaseDate`` property. 
+
+.. code-block:: html
+
+  <form action="/movies/Create" method="post">
+    <div class="form-horizontal">
+      <h4>Movie</h4>
+      <hr />
+      <div class="text-danger"></div>	
+      <div class="form-group">
+        <label class="col-md-2 control-label" for="ReleaseDate">ReleaseDate</label>
+        <div class="col-md-10">
+          <input class="form-control" type="datetime" 
+          data-val="true" data-val-required="The ReleaseDate field is required." 
+          id="ReleaseDate" name="ReleaseDate" value="" />		
+          <span class="text-danger field-validation-valid" 
+          data-valmsg-for="ReleaseDate" data-valmsg-replace="true" />
+        </div>
+      </div>	  
+      </div>
+  </form>
+            
+Client-side validation prevents submission until the form is valid. The Submit button runs JavaScript that either submits the form or displays error messages. 
+
+IClientModelValidator
+---------------------
+You may create client side logic for your custom attribute, and JQuery's unobtrusive validation will execute it on the client for you automatically as part of validation. The first step is to control what values render in the ``data-`` attributes by implementing the ``IClientModelValidator`` interface as shown here:  
```

Comment:
> JQuery's unobtrusive validation

The unobtrusive validation library (jquery.validate.unobtrusive.js) is actually provided by MVC, not jQuery. It's just built on top of jQuery. So the possessive should not be used here.